### PR TITLE
feat: use brotli instead of gzip for compression in NGINX

### PR DESCRIPTION
**Pull request by andreassteinmann**
## PR Type

[x] Feature

## What Is the Current Behavior?
gzip is used to compress files on NGINX.

## What Is the New Behavior?
[ngx_brotli](https://github.com/google/ngx_brotli) is used to compress files on NGINX because Brotli has a better compression ratio compared to gzip.
Two new NGINX modules _ngx_http_brotli_filter_module.so_ (compressing responses on-the-fly) and _ngx_http_brotli_static_module.so_ (serving pre-compressed files) are built in the Dockerfile and referenced in nginx.conf.

## Does this PR Introduce a Breaking Change?

[x] No

## Other Information

[AB#94548](https://dev.azure.com/intershop-com/cefd1005-00a7-4c79-927f-a16947d1b2e6/_workitems/edit/94548)

### CODE REVIEW

**Review by shauke — APPROVED** [review]

**Comment by shauke** [comment]
@intershop/c3po-documentation-l10n can you please review the documentation parts, since a bigger part documentation was added

File: docs/guides/migrations.md [hunk 1]
```diff
@@ -10,28 +10,30 @@ kb_sync_latest_only
...
-To keep the current behavior in an existing project either adapt the `0` default in `pagelet.numberParam('NavigationDepth', 0)"` to a reasonable number or set the 'Navigation Depth' values for all 'Static Content Page' component instances in the ICM backend to reasonable depth values instead of leaving them empty.
+With the content model adaptions of `icm-as-customization-headless:1.7.0`, a depth default value of `5` has been introduced and the description has been changed accordingly.
+In the PWA, the rendering was adapted as well so that an empty `NavigationDepth` value of the 'Static Content Page' component now defaults to `0` instead of no depth limitation that resulted in the whole content page tree being fetched and saved to the state.
+To keep the current behavior in an existing project, either adapt the `0` default in `pagelet.numberParam('NavigationDepth', 0)"` to a reasonable number or set the 'Navigation Depth' values for all 'Static Content Page' component instances in Intershop Commerce Management to reasonable depth values instead of leaving them empty.
```
Comment (mglatter) [inline docs/guides/migrations.md:34]: Should 'Navigation Depth' and 'Static Content Page' also be formatted like `NavigationDepth` and `Static Content Page` or was it done differently here on purpose? (Also see line 33.)

Comment (shauke) [reply docs/guides/migrations.md]: This was done intentionally. I use the code formatting if it is an actual value or name in the code, but when it is a display name in the ICM backoffice I used the single quotes. I could change it to double quotes if this helps anything?

File: docs/guides/nginx-startup.md [hunk 2]
```diff
@@ -255,21 +255,21 @@ An additional Redis service is not intended for production environments where a
...
 
 The two NGINX modules
 
-- ngx_http_brotli_filter_module.so – for compressing responses on-the-fly
-- ngx_http_brotli_static_module.so - for serving pre-compressed files
+- `ngx_http_brotli_filter_module.so` – for compressing responses on-the-fly
+- `ngx_http_brotli_static_module.so` - for serving pre-compressed files
 
-are built in the [Dockerfile](../../nginx/Dockerfile) using an NGINX archive in a version which matches the openresty Docker image version and referenced in [nginx.conf](../../nginx/nginx.conf).
+are built in the [`Dockerfile`](../../nginx/Dockerfile) using an NGINX archive in a version which matches the openresty Docker image version and referenced in [`nginx.conf`](../../nginx/nginx.conf).
```
Comment (mglatter) [inline docs/guides/nginx-startup.md:272]: NGINX is lowercase elsewhere in this document (e.g., "nginx container"). Should it always be capitalized?

**Review by mglatter — DISMISSED** [review]

**Review by shauke — APPROVED** [review]

**Pull request merged**